Updated Eventing Operations Guides (#14734)
kyma-project/kyma: docs/04-operation-guides/troubleshooting/evnt-01-eventing-troubleshooting.md

```bash
kubectl -n kyma-system get eventingBackend eventing-backend -o yaml
```

### Step 2: Check the status of the Subscription

Check the Subscription, if it is ready. Run the command:

```bash
kubectl -n {NAMESPACE} get subscription {NAME}
```
If the Subscription is not ready, check the exact reason of the error in the status of the Subscription by running the command:

```bash
kubectl -n {NAMESPACE} get subscription {NAME} -o yaml
```

If there is a problem with the Subscription sink you will see the message that sink is not valid cluster local svc. 
Verify if the sink is a valid HTTP endpoint, for example: `test.test.svc.cluster.local`.

Check if the Event type defined the Subscription is correctly formatted as specified in the [guidelines](../../05-technical-reference/evnt-01-event-names.md).
Also, check if the Event type is using correct `eventTypePrefix`, the event type has to start with the `eventTypePrefix`, otherwise the Subscription won't be ready.

Run the following command to get the configured `eventTypePrefix` in Eventing Services:
```bash
kubectl get configmaps -n kyma-system eventing -o jsonpath='{.data.eventTypePrefix}'
```

> ### Step 3: Check if the Subscription sink is healthy.

Reviewer: I think we should reorder steps 3, 4 and 5 to respect the flow of data:
- Check the publisher
- Check the messaging server
- Check the dispatcher
- Check the subscriber.

Author: updated

Check the workload Url defined in the Subscription sink is correct and healthy to receive events.
To get the sink from the Subscription, run this command:

```bash
kubectl -n {NAMESPACE} get subscription {NAME} -o jsonpath='{.spec.sink}'
```

You can run the following commands to check the health of the sink:

```bash
kubectl -n default run --image=curlimages/curl --restart=Never sink-test-tmp -- curl --head {SINK_URL}
kubectl -n default logs sink-test-tmp 
kubectl -n default delete pod sink-test-tmp
```

### Step 4: Check the eventing-controller logs.

Check the logs from the Eventing Controller Pod for any errors or to verify that the event is dispatched.
To fetch these logs, run this command:

```bash
kubectl -n kyma-system logs -l app.kubernetes.io/instance=eventing,app.kubernetes.io/name=controller
```

Check for any error messages in the logs. 

> If the event dispatch log `"message":"event dispatched"` is not present for NATS backend, then the issue could be:

Reviewer: If there is a subscriber that cannot process Events, I think we will see that log too but with a none-2XX response status-code. I think we should mention in that case to check the subscriber logs too.

1. The subscriber (i.e. sink) is not reachable.

2. The event was not published in correct format.

3. Eventing-controller is not able to connect to NATS Server.

### Step 5: Check if the Event was published correctly

Check if sending an event returns an HTTP Status Code other than 2xx. 

- If the HTTP Status Code was 4xx, then check if you are sending the Events in correct formats. Eventing supports two Event formats (i.e. legacy and cloud events), see the Eventing [tutorials](../../03-tutorials/00-eventing) section for more information.

- If the HTTP Status Code was 5xx, then check the logs from the Eventing Publisher Proxy Pod for any errors. To fetch these logs, run this command:
    ```bash
    kubectl -n kyma-system logs -l app.kubernetes.io/instance=eventing,app.kubernetes.io/name=eventing-publisher-proxy
    ```
  
- If the HTTP Status Code was 2xx, then verify if you are sending the event on same Event type as defined in the subscription.
  ```bash

>   kubectl -n {NAMESPACE} get subscription {NAME} -o jsonpath='{.spec.filter.filters}' | jq

Reviewer: I think the `jq` part is unnecessary for the troubleshooting itself.

---
title: Kyma Eventing - Basic Diagnostics
---

> ## Symptom(s)

Reviewer 2: ```suggestion
## Symptom
```
It's a [standardized headline](https://raw.githubusercontent.com/kyma-project/community/main/templates/resources/troubleshooting.md).

> 

Author: These are kind of different symptoms but we cannot say anything about the cause of them without doing the steps mentioned in the `Remedy` section. Thats why I covered them in a single doc.

- You publish an event but the event is not received by the subscriber.

- Subscription is not in ready state.

- You are unable to publish events.

## Remedy

Follow these steps to detect the source of the problem:

### Step 1: Check the status of the Eventing backend CR

> Check the Eventing backend CR, if the field `EVENTINGREADY` is true:

Author: I added the numbering back.

   
```bash
kubectl -n kyma-system get eventingbackends.eventing.kyma-project.io
```
If `EVENTINGREADY` is false, check the exact reason of the error in the status of the CR by running the command:

```bash
kubectl -n kyma-system get eventingbackends.eventing.kyma-project.io eventing-backend -o yaml
```

> 

Author: I added a line. Its hard to say exactly what could be an issue if EVENTINGREADY is true.

### Step 2: Check the status of the Subscription

Check the Subscription, if it is ready. Run the command:

```bash
kubectl -n {NAMESPACE} get subscriptions.eventing.kyma-project.io {NAME}
```
If the Subscription is not ready, check the exact reason of the error in the status of the Subscription by running the command:

```bash
kubectl -n {NAMESPACE} get subscriptions.eventing.kyma-project.io {NAME} -o yaml
```

If there is a problem with the Subscription sink you will see the message that sink is not valid cluster local svc. 
Verify if the sink is a valid HTTP endpoint, for example: `test.test.svc.cluster.local`.

Check if the Event type defined the Subscription is correctly formatted as specified in the [guidelines](../../05-technical-reference/evnt-01-event-names.md).
Also, check if the Event type is using correct `eventTypePrefix`, the event type has to start with the `eventTypePrefix`, otherwise the Subscription won't be ready.

Run the following command to get the configured `eventTypePrefix` in Eventing Services:
```bash
kubectl get configmaps -n kyma-system eventing -o jsonpath='{.data.eventTypePrefix}'
```

> ### Step 3: Check if the Event was published correctly

Reviewer 2: ```suggestion
### Step 3: Check if the event was published correctly
```

- If the HTTP Status Code was 4xx, then check if you are sending the Events in correct formats. Eventing supports two Event formats (i.e. legacy and cloud events), see the Eventing [tutorials](../../03-tutorials/00-eventing) section for more information.

- If the HTTP Status Code was 5xx, then check the logs from the Eventing Publisher Proxy Pod for any errors. To fetch these logs, run this command:
    ```bash
    kubectl -n kyma-system logs -l app.kubernetes.io/instance=eventing,app.kubernetes.io/name=eventing-publisher-proxy
    ```

- If the HTTP Status Code was 2xx, then verify if you are sending the event on same Event type as defined in the subscription.
  ```bash
  kubectl -n {NAMESPACE} get subscriptions.eventing.kyma-project.io {NAME} -o jsonpath='{.spec.filter.filters}'
  ```

### Step 4: Check the eventing-controller logs.

Check the logs from the Eventing Controller Pod for any errors or to verify that the event is dispatched.
To fetch these logs, run this command:

```bash
kubectl -n kyma-system logs -l app.kubernetes.io/instance=eventing,app.kubernetes.io/name=controller
```

Check for any error messages in the logs.

> If the event dispatch log `"message":"event dispatched"` is not present for NATS backend, then the issue could be:

Reviewer 2: ```suggestion
If the event dispatch log `"message":"event dispatched"` is not present for NATS backend, the issue could be one of the following:
```

---
title: Kyma Eventing - Basic Diagnostics
---

## Symptom

- You publish an event but the event is not received by the subscriber.

- Subscription is not in ready state.

- You are unable to publish events.

## Remedy

Follow these steps to detect the source of the problem:

### Step 1: Check the status of the Eventing backend CR

1. Check the Eventing backend CR. Is the field **EVENTINGREADY** `true`?
   
    ```bash
    kubectl -n kyma-system get eventingbackends.eventing.kyma-project.io
    ```

2. If `EVENTINGREADY` is false, check the exact reason of the error in the status of the CR by running the command:

    ```bash
    kubectl -n kyma-system get eventingbackends.eventing.kyma-project.io eventing-backend -o yaml
    ```

3. If `EVENTINGREADY` is true, Eventing backend is not an issue. Follow the next steps to find the source of the problem.

### Step 2: Check the status of the Subscription

1. Check whether the Subscription is ready. Run the command:

    ```bash
    kubectl -n {NAMESPACE} get subscriptions.eventing.kyma-project.io {NAME}
    ```

> 2. If the Subscription is not ready, check the exact reason of the error in the status of the Subscription by running the command:

Author: hmm..step 2.3 would not be relevant, but 2.4 would be relevant.

> 2. If the Subscription is not ready, check the exact reason of the error in the status of the Subscription by running the command:

Reviewer 2: So step 3 only applies if the Subscription is not ready? Then it's basically a substep to step 2.2, because readers see that message after running the line 48 command? So we could drop the number `3. ` there and 2.4 would be the new 2.3.

Basically, line 51 would be:

> [3 blank spaces] If the status of the Subscription informs you that the sink is not a valid cluster local svc,  investigate the Subscription sink. Verify if the sink is a ...

...and line 54 would be:

> 3. If the Subscription **_is_** ready, check if the event type...

> 2. If the Subscription is not ready, check the exact reason of the error in the status of the Subscription by running the command:

Author: I have updated step 2.3 (old line 53). For line 54, it is valid for both cases if the Subscription is ready or not.

   - The event was published in a wrong format.

   - Eventing Controller cannot connect to NATS Server.

### Step 5: Check if the Subscription sink is healthy.

1. Check whether the workload URL defined in the Subscription sink is correct and healthy to receive events. To get the sink from the Subscription, run this command:

    ```bash
    kubectl -n {NAMESPACE} get subscriptions.eventing.kyma-project.io {NAME} -o jsonpath='{.spec.sink}'
    ```

2. To check the health of the sink, run the following commands:

    ```bash
    kubectl -n default run --image=curlimages/curl --restart=Never sink-test-tmp -- curl --head {SINK_URL}
    kubectl -n default logs sink-test-tmp 
    kubectl -n default delete pod sink-test-tmp
    ```

    If the returned HTTP status code is not 2xx, check the logs of the subscriber instance.

### Step 6: Check NATS JetStream status

1. Check the health of NATS Pods. Run the command:

    ```bash
    kubectl -n kyma-system get pods -l nats_cluster=eventing-nats
    ```

2. Check if the stream and consumers exist in NATS JetStream by following the [JetStream troubleshooting guide](evnt-02-jetstream-troubleshooting.md).

> 

Reviewer 2: What do we suggest if all checks gave no clear result? Or are we sure that these diagnostic steps will uncover all potential issues?

Reviewer 2: That sounds fine, thank you!
